src/node/desktop/src/main/detect_r.ts

```diff
@@ -37,116 +42,98 @@ function showRNotFoundError(msg: string): void {
-  assert((process.platform === 'win32'));
+}
 
-  // TODO: this is just a hacked-together placeholder for dev purposes
+async function detectREnvironment(): Promise<REnvironment> {
 
-  // check for which R override
-  let rWhichRPath = new FilePath();
-  const whichROverride = getenv('RSTUDIO_WHICH_R');
-  if (!whichROverride) {
-    dialog.showErrorBox(
-      'RSTUDIO_WHICH_R Not Set',
-      'Temporary: RSTUDIO_WHICH_R environment variable must point to R installation folder. Cannot continue.');
-    return false;
+  // scan for R
+  const R = await scanForR();
+  if (!R) {
+    throw rNotFoundErrorMessage();
   }
 
-  rWhichRPath = new FilePath(whichROverride);
-  if (!rWhichRPath.existsSync()) {
-    dialog.showErrorBox(
-      'RSTUDIO_WHICH_R Incorrect',
-      `RSTUDIO_WHICH_R environment variable set incorrectly; ${rWhichRPath.getAbsolutePathNative()} not found. Cannot continue.`);
-    return false;
-  }
+  // get R_HOME + other related R environment variables
+  const stdout = execSync(`${R} RHOME`, { encoding: 'utf-8' });
+  const home = stdout.trim();
+  const envvars = {
+    R_HOME:        `${home}`,
+    R_SHARE_DIR:   `${home}/share`,
+    R_INCLUDE_DIR: `${home}/include`,
+    R_DOC_DIR:     `${home}/doc`
+  };
 
+  // get R version string
+  const rVersion = execSync(
+    `${R} --vanilla -s -e "cat(format(getRversion()))"`,
+    { encoding: 'utf-8' }
+  );
 
-  process.env.R_HOME = rWhichRPath.getAbsolutePathNative();
-  const rPath = rWhichRPath.completeChildPath('bin\\x64');
-  process.env.PATH = `${rPath};${process.env.PATH}`;
+  return {
+    rScriptPath: R,
+    version:     rVersion,
+    envVars:     envvars
+  };
 
-  return true;
 }
 
-async function scanForR(rstudioWhichR: FilePath): Promise<FilePath> {
+async function scanForR(): Promise<string> {
+
+  // if the RSTUDIO_WHICH_R environment variable is set, use that
+  const rstudioWhichR = getenv('RSTUDIO_WHICH_R');
+  if (rstudioWhichR) {
+    logger().logDiagnostic(`Using RSTUDIO_WHICH_R: ${rstudioWhichR}`)
+    return rstudioWhichR;
+  }
+
+  // otherwise, use platform-specific lookup strategies
   if (process.platform === 'win32') {
-    return await scanForRWin32(rstudioWhichR);
+    return await scanForRWin32();
   } else {
-    return await scanForRPosix(rstudioWhichR);
+    return await scanForRPosix();
   }
 }
 
-async function scanForRPosix(rstudioWhichR: FilePath): Promise<FilePath> {
-  assert((process.platform !== 'win32'));
+async function scanForRPosix(): Promise<string> {
 
-  // prefer RSTUDIO_WHICH_R
-  if (!rstudioWhichR.isEmpty()) {
-    return rstudioWhichR;
-  }
-
-  // first look for R on the PATH
-  try {
-    const { stdout } = await asyncExec('/usr/bin/which R', { encoding: 'utf-8' });
-    const R = stdout.trim();
-    if (R) {
-      return new FilePath(R);
-    }
-  } catch (error) {
-    logger().logError(error);
+  // first, look for R on the PATH
+  const stdout = execSync('/usr/bin/which R', { encoding: 'utf-8' });
```

Comment:
This can throw an exception. Non-trivial synchronous node APIs should be assumed to throw, and the docs mention it specifically for this one:
> If the process times out or has a non-zero exit code, this method will throw. The Error object will contain the entire result from child_process.spawnSync().

So you can either handle it internally here, or let the caller deal with it.